plugin-c/crypto.cpp

```diff
@@ -27,6 +28,30 @@ struct string *hook_KRYPTO_sha512(struct string *str) {
   return hexEncode(digest, sizeof(digest));
 }
 
+struct string *hook_KRYPTO_sha512_256raw(struct string *str) {
+  unsigned char digest[32];
+  unsigned int digest_len = 32;
+  EVP_MD_CTX *ctx = EVP_MD_CTX_new();
+  assert(ctx != NULL);
+  assert(EVP_DigestInit_ex(ctx, EVP_sha512_256(), NULL)              == 1);
+  assert(EVP_DigestUpdate(ctx, (unsigned char *)str->data, len(str)) == 1);
```

Comment:
you shouldn't need the cast to `unsigned char *` because the parameter for this function is of type `void *`